### PR TITLE
Static lib and dynamic lib building options

### PR DESCRIPTION
**Pull request by IceDragon200**
Hello, I have returned with another set of minor changes to the cmake file.
Today I propose, `BUILD_DYNAMIC_LIBS` and `SOUNDIO_STATIC_LIBNAME`.

`BUILD_DYNAMIC_LIBS`, allows users to opt in for building the dynamic lib (.so, .dylib, .dll or whatever platforms do these days), it defaults to ON by default, so no one will feel any disturbances (hopefully)

The second option is `SOUNDIO_STATIC_LIBNAME`, since the output for both the static and dynamic libs have the same basename (libsoundio.(a|so)), it can cause some issues with compilers who refuse to be nice and just link the static object first, while this can be alleviated by simply specifying the full path to the static lib in question, in some cases it's out of the question where the compilation process is handled by something else which only offers you the ability link libraries (by name).

tl;dr I can haz `libsoundio_static.a` instead of `libsoundio.a`, to help ease the pain, because no one likes monkeying around their compiler for hours

### CODE REVIEW

**Comment by andrewrk** [comment]
Happy to merge this. Quick question though. If you disable the dynamic library then there will only be libsoundio.a and there will be no libsoundio.so file to cause problems. Perhaps you only need the ability to disable the shared library?

**Comment by IceDragon200** [comment]
@andrewrk In case someone wants to install both libs to their system, they can override the static lib name on for the system. Otherwise I'd just rename the static lib directly to soundio_static, as most other libs I've worked with do that if they offer both

**Comment by andrewrk** [comment]
Thanks for the explanation. I'll accept the pull request as is.
